src/Shouldly/Internals/Is.cs

```diff
@@ -47,15 +47,15 @@ public static bool Equal<T>(IEnumerable<T>? actual, IEnumerable<T>? expected)
             return Equal(actual, expected, GetEqualityComparer<T>());
         }
 
-        public static bool Equal<T>(IEnumerable<T> actual, IEnumerable<T> expected, IEqualityComparer<T> comparer)
+        public static bool Equal<T>(IEnumerable<T>? actual, IEnumerable<T>? expected, IEqualityComparer<T> comparer)
         {
             if (actual == null && expected == null)
                 return true;
             if (actual == null || expected == null)
                 return false;
 
-            var expectedEnum = expected.GetEnumerator();
-            var actualEnum = actual.GetEnumerator();
+            using var expectedEnum = expected.GetEnumerator();
```

Comment:
Disposing is super important. Otherwise e.g. `finally` or `using` blocks in iterator methods never complete, or e.g. streaming database results leak the connection.